internal/provider/user_data_source.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &userDataSource{}
+
+func NewuserDataSource() datasource.DataSource {
+	return &userDataSource{}
+}
+
+// userDataSource defines the data source implementation.
+type userDataSource struct {
+	apiToken string
+}
```

Comment:
I'm not sure I understand what this is doing. Why do we need to define an `apiToken` here?